### PR TITLE
add client js library, adapt imgdetect demo to use client js library - #242

### PR DESCRIPTION
**Pull request by alx**
JS mini client for prediction + updated demo

### CODE REVIEW

File: clients/js/.gitignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+node_modules/*
+.DS_Store
+npm-debug.log
```
Comment (beniz) [inline clients/js/.gitignore]: Please remove this file

File: clients/js/README.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+*This will work in node.js and browsers via [Browserify](http://browserify.org/)*
+
+```js
+var DeepDetect = require('deepdetect-js');
+var app = new DeepDetect.App('http://localhost/api');
+
+```
+
+You can also use the SDK by adding this script to your HTML:
```
Comment (beniz) [inline clients/js/README.md]: This is the easiest usage, can we put it first, i.e. before the installation option via npm, and also separate the two (make two sub-section). Thanks!

File: clients/js/package.json [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    "babel-preset-es2015": "^6.22.0",
+    "eslint": "^3.14.1",
+    "eslint-loader": "^1.6.1",
+    "mocha": "^3.2.0",
+    "webpack": "^2.2.1"
+  },
+  "dependencies": {
+    "axios": "^0.15.3"
+  }
+}
```
Comment (beniz) [inline clients/js/package.json]: Are we really using all these dependencies ? babel, eslint, mocha, webpack, axios ?

File: clients/js/spec/node-test.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+var app;
+
+beforeEach(function() {
+  app = new DeepDetect.App(
+    process.env.API_ENDPOINT
+  );
+});
+
+describe('DeepDetect JS SDK', function() {
```
Comment (beniz) [inline clients/js/spec/node-test.js]: Let's call this the DeepDetect JS client

File: clients/js/src/App.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
+let Predict = require('./Predict');
+let Info = require('./Info');
+let {ERRORS} = require('./constants');
+
+/**
+* top-level class that allows access to predict
+* @class
+*/
+
+class App {
```
Comment (beniz) [inline clients/js/src/App.js]: Change `App` class into `Service` class, to match DD API object names.

File: clients/js/src/Info.js [hunk 6]
```diff
@@ -0,0 +1,18 @@
+let axios = require('axios');
+let {API} = require('./constants');
+
+class Predict {
```
Comment (beniz) [inline clients/js/src/Info.js]: This should be `class Info`

File: clients/js/src/constants.js [hunk 7]
```diff
@@ -0,0 +1,14 @@
+module.exports = {
+  API: {
+    INFO: '/info',
+    SERVICES: '/services',
+    TRAIN: '/train',
```
Comment (beniz) [inline clients/js/src/constants.js]: If we don't have support for train within the client at the moment, let's not put it in here.

File: clients/js/src/helpers.js [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+    });
+    return copy;
+  },
+  checkType: (regex, val)=> {
+    if ((regex instanceof RegExp) === false) {
+      regex = new RegExp(regex);
+    }
+    return regex.test(Object.prototype.toString.call(val));
+  }
+};
```
Comment (beniz) [inline clients/js/src/helpers.js]: Are we using this somewhere ?

File: clients/js/src/index.js [hunk 9]
```diff
@@ -0,0 +1,7 @@
+let App = require('./App');
```
Comment (beniz) [inline clients/js/src/index.js]: Let's change `App` into `Service`.

File: clients/js/yarn.lock [hunk 10]
```diff
@@ -0,0 +1,2921 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
Comment (beniz) [inline clients/js/yarn.lock]: What is this file, do we need it ?

Comment (alx) [inline clients/js/yarn.lock]: https://yarnpkg.com/docs/yarn-lock/#toc-check-into-source-control

> All yarn.lock files should be checked into source control (e.g. git or mercurial). This allows Yarn to install the same exact dependency tree across all machines, whether it be your coworker’s laptop or a CI server.

File: clients/js/src/helpers.js (hunk 8 above)
Comment (alx) [inline clients/js/src/helpers.js]: ```helpers.js``` unused and removed.

File: clients/js/src/constants.js (hunk 7 above)
Comment (alx) [inline clients/js/src/constants.js]: removed.

File: clients/js/src/Info.js (hunk 6 above)
Comment (alx) [inline clients/js/src/Info.js]: renamed

File: clients/js/src/App.js (hunk 5 above)
Comment (alx) [inline clients/js/src/App.js]: renamed.

File: clients/js/spec/node-test.js (hunk 4 above)
Comment (alx) [inline clients/js/spec/node-test.js]: modified.

File: clients/js/package.json (hunk 3 above)
Comment (alx) [inline clients/js/package.json]: The only real dependency is '[axios](https://www.npmjs.com/package/axios)' for promise based http requests.

The devDependencies are necessary to compile (babel, webpack), test (mocha) and keep the code clean (eslint) in the next iterations.

File: clients/js/README.md (hunk 2 above)
Comment (alx) [inline clients/js/README.md]: modified.

File: clients/js/.gitignore (hunk 1 above)
Comment (alx) [inline clients/js/.gitignore]: File necessary to avoid pushing large `node_modules` folder inside the repository.

File: clients/js/.babelrc [hunk 11]
```diff
@@ -0,0 +1,4 @@
+{
+  "presets": ["es2015"],
+  "sourceMaps": false
+}
```
Comment (alx) [inline clients/js/.babelrc]: To translate recent js code in a code that'll be understood by older browsers: https://babeljs.io/

File: clients/js/.eslintrc [hunk 12]
```diff
@@ -0,0 +1,23 @@
...
+    "camelcase": [2, {"properties": "never"}],
+    "new-cap": 2,
+    "no-const-assign": 2,
+    "eqeqeq": 2,
+    "no-multi-str": 2,
+    "react/display-name": 0,
+    "react/prop-types": 0,
+    "react/no-did-mount-set-state": 0
+  }
+}
```
Comment (alx) [inline clients/js/.eslintrc]: To keep the code clean and consistent: http://eslint.org/docs/user-guide/getting-started

I've removed the unused react rules.

File: clients/js/.babelrc (hunk 11 above)
Comment (beniz) [inline clients/js/.babelrc]: Can you add a `make` section on how to build the lib? tks.

Comment (alx) [inline clients/js/.babelrc]: done

**Pull request closed without merging**